pkg/domain/infra/abi/containers.go

```diff
@@ -128,9 +128,28 @@ func getContainers(runtime *libpod.Runtime, options getContainersOptions) ([]con
 		return containers, nil
 	}
 
-	containers := make([]containerWrapper, len(libpodContainers))
-	for i := range libpodContainers {
-		containers[i] = containerWrapper{Container: libpodContainers[i], rawInput: libpodContainers[i].ID()}
+	// If no names or IDs are specified, we can return the result as is.
+	// Otherwise, we need to do some further lookups.
+	if len(options.names) == 0 || (len(options.names) == 1 && options.names[0] == "") {
```

Comment:
Why the check for `options.names[0] == ""`, shouldn't the caller not pass in the name in the slice when it is empty?

Reply:
In an ideal world yes but most callers just do `[]string{nameOrID}`.  I prefer to play it safe here as I am afraid of subtle/silent regressions in the future.

What I plan on doing is to extend `podman ps` to also accept name/ID arguments.  Then all this scattered logic should go into `./libpod/`.